.github/workflows/nnf-rpm.yml

```diff
@@ -0,0 +1,121 @@
+name: switchtec-user rpm
+
+on: push
+
+env:
+  STARTS_WITH_RELEASES: 'refs/heads/releases/nnf'
+
+jobs:
+  build-rpm:
+    runs-on: ubuntu-latest
+    container:
+      image: centos:8
+      options: --cpus 1
+
+    steps:
+      - name: "Verify context"
+        run: |
+          echo "ref is ${{ github.ref }}"
+          echo "ref_type is ${{ github.ref_type }}"
+          echo "event_type is ${{ github.event.event_type }}"
+
+      - name: centos is EOL so muck with it
+        run: |
+          sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-Linux-*
+          sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-Linux-*
+      - name: install tools
+        # The "Development Tools" group satisfies everything, but it's a big
+        # hammer.
+        #   run: dnf -y groupinstall "Development Tools"
+        #
+        # So let's cherry-pick:
+        run: dnf install -y rpm-build rpmdevtools rpmlint make gcc git ncurses-devel openssl-devel
+      - name: personalize rpmlint
+        # suppress one error: we want to use /usr/local.
+        run: |
+          mkdir ~/.config
+          echo 'addFilter("dir-or-file-in-usr-local")' > ~/.config/rpmlint
+
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          # Because we need the tags to build the version ID.
+          fetch-depth: 0
+
+      - name: generate the version ID
+        # The VERSION-GEN tool can generate a version ID that includes the
+        # nearest tag, the number of commits since that tag, the short hash of
+        # the latest commit, and whether or not the workarea is dirty.
+        #
+        # The version ID will look like this, with the "-dirty" suffix removed:
+        #   4.0-rc1-31-gbb58f3e
+        # (The "-dirty" suffix happens because we're modifying the ./version
+        # file when we do this.)
+        run: |
+          rm -f version
+          ./VERSION-GEN
+          awk '{print $3}' doc/version | sed -e 's/"//g;' -e 's/\-dirty$//' > version
+          cat version
+          echo "versionDotted=$(<version)" >> $GITHUB_ENV
+          echo "versionDashed=$(sed 's/\./-/g' version)" >> $GITHUB_ENV
+
+      - name: configure the source
+        run: ./configure
+      - name: make source tarball
+        run: make dist
+      - name: view source artifacts
+        run: /bin/ls -al *.xz *.spec
+      - name: view the spec file
+        run: cat *.spec
+      - name: lint the spec file
+        run: rpmlint ./switchtec.spec
```

Comment:
This step generates a warning, should that fail?
```
Run rpmlint ./switchtec.spec
./switchtec.spec: W: invalid-url Source0: switchtec-4.0-0.rc1.15.g8a4839d.tar.xz
0 packages and 1 specfiles checked; 0 errors, 1 warnings.
```

Reply:
If there were errors, then it would exit non-zero, and fail the workflow.

This PR fixes all of the errors in the spec file.